.changeset/unlucky-camels-smile.md

```diff
@@ -0,0 +1,5 @@
+---
+"ts-essentials": minor
```

Comment:
Given the convention that I've just added, it doesn't break the backwards compatibility so I'm happy to continue with minor

Although, if it's treating as a break of contract, it's changing the current behaviour (type error => no type error) therefore I'm also happy to make it major

Any input from anyone?

Reply:
After @Nightbr's comment, I've considered using major instead of minor - https://github.com/ts-essentials/ts-essentials/issues/343#issuecomment-2041623070